nil/services/synccommittee/prover/tracer/utils.go

```diff
@@ -17,3 +17,16 @@ func getDataOverflowSafe(data []byte, start uint64, size uint64) []byte {
 	}
 	return common.RightPadBytes(data[start:end], int(size))
 }
+
+// getDataIfStartValid returns a slice from `data` starting at `start` with a length of `size` bytes.
+// It returns nil if `size` is zero or if `start` is out of bounds.
+// Note: It does not check if `start+size` exceeds `data` length.
+func getDataIfStartValid(data []byte, start uint64, size uint64) []byte {
+	if size == 0 {
+		return nil
+	}
+	if len(data) > int(start) {
```

Comment:
`len(data) > int(start + size)`?

Reply:
In this case, the conditions appear to be interchangeable, as the logic underpinning the execution of the EVM remains consistent. If the start is valid, then the end of the range must also be valid; otherwise, the execution would have already failed.
https://github.com/NilFoundation/nil/blob/9d091d81a06cf437a18318fa39308364650699b2/nil/internal/vm/memory.go#L85

Reply:
The [check](https://github.com/NilFoundation/nil/blob/9d091d81a06cf437a18318fa39308364650699b2/nil/internal/vm/memory.go#L85) also looks strange.
If assume that `size` was checked before, condition `len(data) > int(start + size)` more strict than required. At the same time not leads over calculation and looks better for understand.